### DISCUSSION TITLE
fatal error LNK1181: cannot open input file 'C:\carla\carla\Unreal\CarlaUE4\Plugins\Carla\CarlaDependencies\lib\osm2odr.lib' make: *** [CarlaUE4Editor] Error 6

### DISCUSSION THREAD

**Discussion by alvaro-s29** [carla-simulator/carla#6918] [General]
Windows 11 Home
Carla 0.9.14
VS 2019
Python 3.11

Hello, Greetings to all
I am having problems since the Make PythonAPI with the OSM2ODR.h file: 

LINK : fatal error LNK1181: cannot open input file 'C:carlaUE4Editor'.
make: *** [CarlaUE4Editor] Error 6

Could you help me. It's for a university research project

Gracias

**Comment by juanrodriguezdekra** [comment]
Hello, 
I am having problems with the same file osm2odr.lib and OSM2ODR.h
The error results in the final make launch step with the following code:


C:\carla>make launch
-[BuildUE4Plugins]: [Batch params]: --build --all
HEAD is now at 260273d Correct descriptor
-[BuildUE4Plugins]: STREETMAP has been successfully installed in "C:\carla\Unreal\CarlaUE4\Plugins\"
-[Setup]: [Batch params]: --boost-toolset msvc-14.2 --generator "" --all
-[Setup]: Asynchronous jobs:  8
-[Setup]: Boost toolset:      msvc-14.2
-[Setup]: Generator:          "Visual Studio 16 2019"
-[Setup]: Install directory:  "C:\carla\Build\"
-[Setup]: Installing zlib...
    -[install_zlib]: [Batch params]: --build-dir "C:\carla\Build\"
    -[install_zlib]: A zlib installation already exists.
    -[install_zlib]: Delete "C:\carla\Build\zlib-install\" if you want to force a rebuild.
    -[install_zlib]: Exiting...
        1 archivo(s) copiado(s).
-[Setup]: Installing libpng...
    -[install_libpng]: [Batch params]: --build-dir "C:\carla\Build\" --zlib-install-dir "C:\carla\Build\zlib-install\"
    -[install_libpng]: A libpng installation already exists.
    -[install_libpng]: Delete "C:\carla\Build\libpng-1.2.37-install\" if you want to force a rebuild.
    -[install_libpng]: Exiting...
-[Setup]: Installing rpclib...
    -[install_rpclib]: [Batch params]: --build-dir "C:\carla\Build\" --generator "Visual Studio 16 2019"
    -[install_rpclib]: A rpclib installation already exists.
    -[install_rpclib]: Delete "C:\carla\Build\rpclib-install\" if you want to force a rebuild.
    -[install_rpclib]: Exiting...
-[Setup]: Installing Google Test...
    -[install_gtest]: [Batch params]: --build-dir "C:\carla\Build\" --generator "Visual Studio 16 2019"
    -[install_gtest]: A Google Test installation already exists.
    -[install_gtest]: Delete "C:\carla\Build\gtest-install\" if you want to force a rebuild.
    -[install_gtest]: Exiting...
-[Setup]: Installing "Recast & Detour"...
    -[install_recast]: [Batch params]: --build-dir "C:\carla\Build\" --generator "Visual Studio 16 2019"
    -[install_recast]: A "Recast & Detour" installation already exists.
    -[install_recast]: Delete "C:\carla\Build\recast-install\" if you want to force a rebuild.
    -[install_recast]: Exiting...
-[Setup]: Installing Boost...
    -[install_boost]: [Batch params]: --build-dir "C:\carla\Build\" --toolset msvc-14.2 --version 1.80.0 -j 8
    -[install_boost]: A Boost installation already exists.
    -[install_boost]: Delete "C:\carla\Build\boost-1.80.0-install\" if you want to force a rebuild.
    -[install_boost]: Exiting...
-[Setup]: Installing Xercesc...
    -[install_xercesc]: [Batch params]: --build-dir "C:\carla\Build\" --generator "Visual Studio 16 2019"
    -[install_xercesc]: A xerces installation already exists.
    -[install_xercesc]: Delete "C:\carla\Build\xerces-c-3.2.3-install\" if you want to force a rebuild.
    -[install_xercesc]: Exiting...
        1 archivo(s) copiado(s).
        1 archivo(s) copiado(s).
-[Setup]: Installing Sqlite3
    -[install_sqlite3]: [Batch params]: --build-dir "C:\carla\Build\"
    -[install_sqlite3]: A Sqlite3 installation already exists.
    -[install_sqlite3]: Delete "C:\carla\Build\sqlite3-install" if you want to force a rebuild.
    -[install_sqlite3]: Exiting...
        1 archivo(s) copiado(s).
        1 archivo(s) copiado(s).
-[Setup]: Installing PROJ
    -[install_proj]: [Batch params]: --build-dir "C:\carla\Build\" --generator "Visual Studio 16 2019"
    -[install_proj]: A PROJ installation already exists.
    -[install_proj]: Delete "C:\carla\Build\proj-install" if you want to force a rebuild.
    -[install_proj]: Exiting...
        1 archivo(s) copiado(s).
        1 archivo(s) copiado(s).
-[Setup]: Installing Eigen
    -[install_eigen]: [Batch params]: --build-dir "C:\carla\Build\"
0 archivo(s) copiado(s)
0 archivo(s) copiado(s)
        1 archivo(s) copiado(s).
        1 archivo(s) copiado(s).

    -[install_eigen]: Eigen has been successfully installed in "C:\carla\Build\eigen-install"
    -[install_eigen]: Exiting...
-[Setup]: Creating "CMakeLists.txt.in"...
-[Setup]:
   ###########
   # SUCCESS #
   ###########

   IMPORTANT

   All the CARLA library dependences should be installed now.
   (You can remove all "*-src" folders in C:\carla\Build\ directory)

   You only need the ASSET PACK with all the meshes and textures.

   This script provides the assets for CARLA Latest:
   You can download the assets from here:

       http://carla-assets.s3.amazonaws.com/20231108_c5101a5.tar.gz

   Unzip it in the "C:\carla\Unreal\CarlaUE4\Content\Carla\" folder.
   If you want another version, search it in C:\carla\Util\ContentVersions.txt.

-[BuildLibCarla]: [Batch params]: --server --client --generator ""
-- Selecting Windows SDK version 10.0.20348.0 to target Windows 10.0.19042.
-- Build debug:   ON
-- Build release: ON
-- Build test:    ON
-- Configuring done (0.0s)
-- Generating done (0.0s)
-- Build files have been written to: C:/carla/Build/libcarla-visualstudio
Microsoft (R) Build Engine versión 16.11.2+f32259642 para .NET Framework
Copyright (C) Microsoft Corporation. Todos los derechos reservados.

  1>Checking Build System
  carla_server.vcxproj -> C:\carla\Build\libcarla-visualstudio\LibCarla\cmake\server\Release\carla_server.lib
  carla_server_debug.vcxproj -> C:\carla\Build\libcarla-visualstudio\LibCarla\cmake\server\Release\carla_server_debug.lib
  1>
  -- Install configuration: "Release"
-- Selecting Windows SDK version 10.0.20348.0 to target Windows 10.0.19042.
-- Build debug:   ON
-- Build release: ON
-- Build test:    ON
-- Configuring done (0.0s)
-- Generating done (0.1s)
-- Build files have been written to: C:/carla/Build/libcarla-visualstudio
Microsoft (R) Build Engine versión 16.11.2+f32259642 para .NET Framework
Copyright (C) Microsoft Corporation. Todos los derechos reservados.

  1>Checking Build System
  carla_client.vcxproj -> C:\carla\Build\libcarla-visualstudio\LibCarla\cmake\client\Release\carla_client.lib
  carla_client_debug.vcxproj -> C:\carla\Build\libcarla-visualstudio\LibCarla\cmake\client\Release\carla_client_debug.lib
  1>
  -- Install configuration: "Release"
-[BuildLibCarla]: LibCarla server has been successfully installed in "C:\carla\Unreal\CarlaUE4\Plugins\Carla\CarlaDependencies\"!
-[BuildLibCarla]: LibCarla client has been successfully installed in "C:\carla\PythonAPI\carla\dependencies\"!
-[BuildOSM2ODR]: [Batch params]: --generator "" --build --all
HEAD is now at 1835e1e9538 Remove logs
CMake Warning:
  Ignoring extra path from command line:

   "C:\carla\Build\om2odr-source""


CMake Error: The source directory "C:/carla/Build/osm2odr-visualstudio/x64" does not exist.
Specify --help for usage, or press the help button on the CMake GUI.
Error: could not load cache
El sistema no puede encontrar el archivo especificado.
El sistema no puede encontrar el archivo especificado.
-[BuildOSM2ODR]: OSM2ODR has been successfully installed in "C:\carla\PythonAPI\carla\dependencies\"
-[BuildCarlaUE4]: [Batch params]: --build --all
--build
--all
ECHO está desactivado.
-[BuildCarlaUE4]: Building Unreal Editor...
Using 'git status' to determine working set for adaptive non-unity build (C:\Users\jmrg\UnrealEngine).
Waiting for 'git status' command to complete
Building CarlaUE4Editor...
Using Visual Studio 2019 14.29.30153 toolchain (C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133) and Windows 10.0.18362.0 SDK (C:\Program Files (x86)\Windows Kits\10).
[Upgrade]
[Upgrade] Using backward-compatible build settings. The latest version of UE4 sets the following values by default, which may require code changes:
[Upgrade]     bLegacyPublicIncludePaths = false                 => Omits subfolders from public include paths to reduce compiler command line length. (Previously: true).
[Upgrade]     ShadowVariableWarningLevel = WarningLevel.Error   => Treats shadowed variable warnings as errors. (Previously: WarningLevel.Warning).
[Upgrade]     PCHUsage = PCHUsageMode.UseExplicitOrSharedPCHs   => Set in build.cs files to enables IWYU-style PCH model. See https://docs.unrealengine.com/en-US/Programming/BuildTools/UnrealBuildTool/IWYU/index.html. (Previously: PCHUsageMode.UseSharedPCHs).
[Upgrade] Suppress this message by setting 'DefaultBuildSettings = BuildSettingsVersion.V2;' in CarlaUE4Editor.Target.cs, and explicitly overriding settings that differ from the new defaults.
[Upgrade]
Building 7 actions with 8 processes...
  [1/7] UE4Editor-StreetMapRuntime.dll
  LINK : fatal error LNK1181: no se puede abrir el archivo de entrada 'C:\carla\Unreal\CarlaUE4\Plugins\Carla\CarlaDependencies\lib\osm2odr.lib'
  [2/7] UE4Editor-CarlaUE4.dll
  LINK : fatal error LNK1181: no se puede abrir el archivo de entrada 'C:\carla\Unreal\CarlaUE4\Plugins\Carla\CarlaDependencies\lib\osm2odr.lib'
  [3/7] UE4Editor-Carla.dll
  LINK : fatal error LNK1181: no se puede abrir el archivo de entrada 'C:\carla\Unreal\CarlaUE4\Plugins\Carla\CarlaDependencies\lib\osm2odr.lib'
  [4/7] Module.CarlaTools.cpp
  C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um\winnt.h(603): warning C4005: 'TEXT': redefinición de macro
  C:\Users\jmrg\UnrealEngine\Engine\Source\Runtime\Core\Public\HAL/Platform.h(1082): note: vea la definición anterior de 'TEXT'
  C:\carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla\Vehicle\WheeledVehicleMovementComponentNW.h(120): warning C4996: 'UWheeledVehicleMovementComponent': PhysX is deprecated. Use the UChaosWheeledVehicleMovementComponent fron the ChaosVehiclePhysics Plugin. Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.
  C:\carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla\Vehicle\WheeledVehicleMovementComponentNW.h(148): warning C4996: 'UVehicleWheel': PhysX is deprecated. Use the UChaosVehicleWheel from the ChaosVehiclePhysics Plugin. Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.
  C:\carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla/Vehicle/CarlaWheeledVehicle.h(72): warning C4996: 'AWheeledVehicle': PhysX is deprecated. Use the AWheeledVehiclePawn from the ChaosVehiclePhysics Plugin. Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.
  C:\carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla/Vehicle/CarlaWheeledVehicle.h(188): warning C4996: 'UWheeledVehicleMovementComponent4W': PhysX is deprecated. Use the UChaosWheeledVehicleMovementComponent from the ChaosVehiclePhysics Plugin. Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.

  You have included windows.h before MinWindows.h
  All useless stuff from the windows headers won't be excluded !!!

  C:\Users\jmrg\UnrealEngine\Engine\Source\Runtime\Core\Public\Windows/MinWindows.h(44): warning C4005: 'NOMINMAX': redefinición de macro
  C:\carla\Unreal\CarlaUE4\Plugins\Carla\CarlaDependencies\include\boost/asio/detail/config.hpp(1566): note: vea la definición anterior de 'NOMINMAX'
  C:/carla/Unreal/CarlaUE4/Plugins/CarlaTools/Source/CarlaTools/Private/Online/CustomFileDownloader.cpp(11): fatal error C1083: No se puede abrir el archivo incluir: 'OSM2ODR.h': No such file or directory
make: *** [CarlaUE4Editor] Error 6

**Reply by dighepr** [reply]
@juanrodriguezdekra and @alvaro-s29 Did you find any solution to this error?

**Reply by nikolatopic131** [reply]
Having the same issue. Posting this to get notified if someone comes up with a solution.

**Comment by nikolatopic131** [comment]
Found the solution, here goes:

The issue is in the file BuildOSM2ODR.bat that's under carla\Util\BuildTools.
In line 112
cmake -G %GENERATOR% %PLATFORM%^
should be changed to
cmake -G %GENERATOR% -A x64^

**Reply by shraddha-git-tech** [reply]
already same `cmake -G %GENERATOR% -A x64^` in that file. Any other solution?

**Reply by nikolatopic131** [reply]
Unfortunately I don't know. This was the one that worked for me, haven't researched any further once I got it working. I hope you manage to find a solution though, good luck.
